src/Product/ProductHelper.php

```diff
@@ -254,6 +254,21 @@ public function get_synced_google_product_ids( WC_Product $product ): ?array {
 	 * @return int the ID for the WC product linked to the provided Google product ID (0 if not found)
 	 */
 	public function get_wc_product_id( string $mc_product_id ): int {
+		/**
+		 * Filters the Merchant Center product ID into a {slug}_{id} format before it's converted to
+		 * a WooCommerce product ID.
+		 *
+		 * This is done first to permit possible mapping to a different product ID. For example, map WooCommerce
+		 * product 5678 to Merchant Center product gla_1234.
+		 *
+		 * @param string $mc_product_id Merchant Center product ID.
+		 * @param string $prefix        The slug prefixed to generated Merchant Center product IDs.
+		 * @since x.x.x
+		 *
+		 * @return string Merchant Center product ID as normally generated by the plugin (e.g., gla_1234).
+		 */
+		$mc_product_id = apply_filters( 'woocommerce_gla_get_wc_product_id', $mc_product_id, $this->get_slug() . '_' );
+
```

Comment:
Yeah, I went back and forth on this one. The symmetry of WC > MC and MC > WC seemed nicer to me, but I was also concerned with allowing a possible custom mapping to another GLA-generated ID (i.e. make product 500 sync to product 55, so 500 > gla_55) and somehow landed on the decision that it would be better to place it earlier. Thanks for calling it out, updated in 6f69d09d4ce82cd37898536760b240a73b6b3570

Reply:
I actually think being able to map product 55 to 500 is an advantage. While probably not common, we could end up with a situation where someone migrates to a new site while exporting/importing products. Yet they still want to keep their original campaigns / products in MC.